utils/sysfs/sysfs.go

```diff
@@ -98,7 +98,25 @@ func (self *realSysFs) GetBlockDeviceSize(name string) (string, error) {
 }
 
 func (self *realSysFs) GetNetworkDevices() ([]os.FileInfo, error) {
-	return ioutil.ReadDir(netDir)
+	files, err := ioutil.ReadDir(netDir)
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter out non-directory & non-symlink files
+	var dirs []os.FileInfo
+	for _, f := range files {
+		if f.Mode()|os.ModeSymlink != 0 {
+			f, err = os.Stat(path.Join(netDir, f.Name()))
+			if err != nil {
+				continue
+			}
+		}
```

Comment:
I don't think this works for a normal case where we just have symlinks:
$ ls -l /sys/class/net/
total 0
lrwxrwxrwx 1 root root 0 Feb 14 18:09 docker0 -> ../../devices/virtual/net/docker0
lrwxrwxrwx 1 root root 0 Jan 20 06:58 eth0 -> ../../devices/pci0000:00/0000:00:04.0/virtio1/net/eth0

eth0 will pass the test for symlink and fail the one for IsDir(). We'll end up not adding it to the known devices.